tests_new/integration_tests/modules/catalog/test_glossaries.py

```diff
@@ -0,0 +1,216 @@
+
+
+def test_create_category(client1, category1):
+    assert_that(category1.nodeUri).is_not_none()
+    assert_that(category1.label).is_equal_to('category1')
+
+
+def test_update_category(client1, category1, session_id):
+    response = update_category(
+        client1,
+        node_uri=category1.nodeUri,
+        name=category1.label,
+        read_me=f'UPDATED: {session_id} Category created for integration testing',
+    )
+    assert_that(response.readme).is_equal_to(f'UPDATED: {session_id} Category created for integration testing')
+
+
+def test_delete_category(client1, glossary1):
+    category = create_category(
+        client1, name='glossary1', parent_uri=glossary1.nodeUri, read_me='Category created for integration testing'
+    )
+    response = delete_category(client1, category.nodeUri)
+    assert_that(response).is_true()
+
+
+def test_delete_category_with_terms(client1, glossary1):
+    category = create_category(
+        client1, name='category1', parent_uri=glossary1.nodeUri, read_me='Category created for integration testing'
+    )
+    term = create_term(
+        client1, name='term1', parent_uri=category.nodeUri, read_me='Term created for integration testing'
+    )
+    response = delete_category(client1, node_uri=category.nodeUri)
+    assert_that(response).is_true()
+
+
+def test_create_term_in_glossary(client1, glossary_term1):
+    assert_that(glossary_term1.nodeUri).is_not_none()
+    assert_that(glossary_term1.label).is_equal_to('glos_term1')
+
+
+def test_create_term_in_category(client1, category_term1):
+    assert_that(category_term1.nodeUri).is_not_none()
+    assert_that(category_term1.label).is_equal_to('cat_term1')
+
+
+def test_update_term(client1, glossary_term1, session_id):
+    response = update_term(
+        client1,
+        node_uri=glossary_term1.nodeUri,
+        name=glossary_term1.label,
+        read_me=f'UPDATED: {session_id} Glossary term created for integration testing',
+    )
+    assert_that(response.readme).is_equal_to(f'UPDATED: {session_id} Glossary term created for integration testing')
+
+
+def test_delete_term(client1, group1, category1):
+    term = create_term(
+        client1, name='toDelete', parent_uri=category1.nodeUri, read_me='Term created for integration testing'
+    )
+    response = delete_term(client1, node_uri=term.nodeUri)
+    assert_that(response).is_true()
+
+
+def test_approve_term_association(
+    client1, dataset_association_with_glossary_term1, session_glossary1, session_glossary_term1
+):
+    response = approve_term_association(client1, link_uri=dataset_association_with_glossary_term1.linkUri)
+    assert_that(response).is_true()
+    response = list_glossary_associations(client1, node_uri=session_glossary1.nodeUri)
+    association = next(
+        (n for n in response.associations.nodes if n.linkUri == dataset_association_with_glossary_term1.linkUri), None
+    )
+    assert_that(association.approvedBySteward).is_equal_to(True)
+
+
```

Comment:
approve and dismiss term association we may want to test the unauthorized path - I believe only glossary owner or admin can approve or dismiss via `_verify_term_assocaition_approver_role()`

Reply:
Great catch, you are right, the permissions are checked in the repositories layer!